src/core/plot/qgsvectorlayerplotdatagatherer.cpp

```diff
@@ -0,0 +1,185 @@
+/***************************************************************************
+                         qgsvectorlayerplotdatagatherer.cpp
+                         -----------------
+    begin                : August 2025
+    copyright            : (C) 2025 by Mathieu
+    email                : mathieu at opengis dot ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "moc_qgsvectorlayerplotdatagatherer.cpp"
+#include "qgsvectorlayerplotdatagatherer.h"
+#include "qgsexpression.h"
+
+
+QgsVectorLayerXyPlotDataGatherer::QgsVectorLayerXyPlotDataGatherer( const QgsFeatureIterator &iterator, const QgsExpressionContext &expressionContext, const QList<QgsVectorLayerXyPlotDataGatherer::XySeriesDetails> &seriesDetails, Qgis::PlotAxisType xAxisType, const QStringList &predefinedCategories )
+  : mIterator( iterator )
+  , mExpressionContext( expressionContext )
+  , mXAxisType( xAxisType )
+  , mSeriesDetails( seriesDetails )
+  , mPredefinedCategories( predefinedCategories )
+{
+}
+
+bool QgsVectorLayerXyPlotDataGatherer::run()
+{
+  QStringList gatheredCategories;
+  QList<QMap<QString, double>> gatheredSeriesCategoriesSum;
+  QList<QgsXyPlotSeries *> gatheredSeries;
```

Comment:
These get leaked if when we return false in this function -- I'd use `std::vector< std::unique_ptr< QgsXyPlotSeries > >` instead